language/move-prover/abigen/src/abigen.rs

```diff
@@ -69,39 +71,77 @@ impl<'env> Abigen<'env> {
     /// Generates ABIs for all script modules in the environment (excluding the dependency set).
     pub fn gen(&mut self) {
         for module in self.env.get_modules() {
-            if module.is_script_module() && module.is_target() {
+            if module.is_target() {
                 let mut path = PathBuf::from(&self.options.output_directory);
-                path.push(
-                    PathBuf::from(module.get_source_path())
-                        .with_extension("abi")
-                        .file_name()
-                        .expect("file name"),
-                );
-
-                match self.compute_abi(&module) {
-                    Ok(abi) => {
-                        self.output.insert(path.to_string_lossy().to_string(), abi);
-                    }
-                    Err(error) => panic!(
-                        "Error while processing script file {:?}: {}",
-                        module.get_source_path(),
-                        error
-                    ),
+                // We make a directory for all of the script function ABIs in a module. But, if
+                // it's a script, we don't create a directory.
+                if !module.is_script_module() {
+                    path.push(
+                        PathBuf::from(module.get_source_path())
+                            .file_stem()
+                            .expect("file extension"),
+                    )
+                }
+
+                for abi in self
+                    .compute_abi(&module)
+                    .map_err(|err| {
+                        format!(
+                            "Error while processing file {:?}: {}",
+                            module.get_source_path(),
+                            err
+                        )
+                    })
+                    .unwrap()
+                {
+                    // If the module is a script module, then the generated ABI needs to be a
```

Comment:
Sure but do you need an assert for this? Isn't it clear from the code of `compute_abi` ?

Reply:
It should be totally clear. I'll remove the assert :)